src/solvers/dgmulti/flux_differencing.jl

```diff
@@ -4,16 +4,23 @@
 # See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
 @muladd begin
 
-#   hadamard_sum!(du, A, volume_flux, orientation, u, equations,
-#                              sparsity_pattern, skip_index=(i,j)->false)
+#   hadamard_sum!(du, A, volume_flux, orientation, u, equations, sparsity_pattern)
 #
 # Computes the flux difference ∑_j A[i, j] * f(u_i, u_j) and accumulates the result into `du`.
+# Called by `local_flux_differencing` to compute local contributions to flux differencing
+# volume integrals.
+#
 # - `du`, `u` are vectors
 # - `A` is the skew-symmetric flux differencing matrix.
-# - `sparsity_pattern`: either `nothing` or an AbstractSparseMatrix which specifies the sparsity
-#   pattern of `A`
-@inline function hadamard_sum!(du, A, volume_flux, orientation, u, equations,
-                               sparsity_pattern::Nothing, skip_index=(i,j)->false)
+# - `sparsity_pattern`: an AbstractSparseMatrix which specifies the sparsity pattern of `A`.
+#                       Alternatively, if `sparsity_pattern::Nothing`, then we use assume the
+#                       operator is fully dense.
+
+# Version for symmetric fluxes and dense operators
+@inline function hadamard_sum!(du, A,
+                               flux_is_symmetric::Val{true}, volume_flux,
+                               orientation, u, equations,
+                               sparsity_pattern::Nothing)
```

Comment:
What about moving the `sparsity_pattern` before `A` such that we follow the rule "specifications of something directly before the specified stuff" consistently? That would be in accordance with the order `flux_is_symmetric, volume_flux`. This would require changes wherever `hadamard_sum!` appears.
```suggestion
@inline function hadamard_sum!(du, sparsity_pattern::Nothing, A,
                               flux_is_symmetric::Val{true}, volume_flux,
                               orientation, u, equations)
```
This is only  a suggestions that we might discuss or discard (but it might make Mr. Consistency happy :wink:)

Reply:
I like that suggestion. We used to assign a default value to `sparsity_pattern` so that's why it was at the end, so I like the idea of moving it around.